tests/common/fixtures/consistency_checker/query-asic/query-asic.py

```diff
@@ -0,0 +1,270 @@
+
+    :param query_objects: The deserialized JSON input file format
+    :return: The deserialized JSON output format
+    """
+
+    results = defaultdict(dict)
+
+    for oid_label_key, attributes in query_objects.items():
+        try:
+            logger.debug(f"Querying ASIC for object key {oid_label_key}")
+            oid = get_numeric_oid_from_label(oid_label_key)
+        except Exception as e:
+            err_msg = f"Failed to extract oid from label '{oid_label_key}': {e}"
+            logger.warning(err_msg)
+            for attribute in attributes:
+                results[oid_label_key][attribute] = {"success": False, "error": err_msg, "asicValue": None}
+            continue
+
+        for attribute in attributes:
+            try:
+                logger.debug(f"Querying ASIC object {oid_label_key} ({oid}) for attribute {attribute}")
+                attribute_oid = lookup_attribute_value_in_pysairedis(attribute)
+                asic_value = get_attribute_value_from_asic(oid, attribute_oid)
+
+                # Convert to str to match how values are represented in ASIC_DB
+                if asic_value in [True, False]:
+                    # ASIC_DB represents these as lowercase
+                    asic_value = str(asic_value).lower()
+                elif asic_value is None:
+                    asic_value = "NULL"
+                else:
+                    asic_value = str(asic_value)
+
+                # Success
+                results[oid_label_key][attribute] = {"asicValue": asic_value, "success": True}
+                logger.debug((f"Got ASIC object {oid_label_key} ({oid}) -> attribute {attribute} ({attribute_oid}) "
+                              f"value {asic_value}"))
+
+            except Exception as e:
+                err_msg = f"Failed to lookup attribute '{attribute}': {e}"
+                logger.warning(err_msg)
+                results[oid_label_key][attribute] = {"success": False, "error": err_msg, "asicValue": None}
+
+    return dict(results)
+
+
+def initialize_sai_api():
+    """
+    Initialize the SAI API
+    """
+    logger.info("Initializing SAI API")
+    profileMap = dict()
+    profileMap[pysairedis.SAI_REDIS_KEY_ENABLE_CLIENT] = "true"
+    status = pysairedis.sai_api_initialize(0, profileMap)
+    assert status == pysairedis.SAI_STATUS_SUCCESS, "Failed to initialize SAI API"
+    logger.info("SAI API initialized")
+
+
+def uninitialize_sai_api():
+    """
+    Uninitialize the SAI API
+    """
+    logger.info("Uninitializing SAI API")
+    status = pysairedis.sai_api_uninitialize()
+    assert status == pysairedis.SAI_STATUS_SUCCESS, "Failed to uninitialize SAI API"
+    logger.info("SAI API uninitialized")
+
+
+def main(args):
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description=HELP_TEXT)
+    parser.add_argument("-i", "--input", help="Input JSON file", required=True)
```

Comment:
Non-blocking comment: consider using [argparse's FileType object](https://docs.python.org/3/library/argparse.html#filetype-objects). That can handle `-` to represent stdin.